### ISSUE TITLE
Don't require frames to create left or right indent for systems

### ISSUE THREAD

**Issue by oMrSmith** [musescore/MuseScore#17325]
### Your idea

I want to propose the following feature:

Allow to create a left or right indent for a system by

a) dragging the first or last barline of a system 

b) selecting a system break - I'm mean the layout symbol - and adjusting a new inspector setting
- "right indent for current system"
- "left indent for following system"

### Problem to be solved

Creating an indent in MU is quite laborious and could become
easier and more intuitive, for example:

https://user-images.githubusercontent.com/78870770/233846759-480a691a-95f2-49d7-98e8-66a4ea43e50b.mp4

**Comment by MarcSabatella** [comment]
The main issue we always face in trying to have any sort of system-dependent settings is defining exactly what that applies to, given that systems are constantly being regenerated.  Like, if you apply an left indent to a system that currently contains measures 21-24, what happens if the layout later changes and 21 is no longer the start of the system?  If the new layout has systems containing 19-22 and another 23-26, which of those should get the indent - or both, or neither?

So while slightly non-intuitive, frames have the advantage of being a very specific permanent element with clearly defined semantics as layout changes.  This is also why vertical spacing is done with spacers instead of properties.

That said, coming up with a comprehensive framework for defining how to handle these sorts of things would be great someday indeed.  Ideally it would also not only indents and vertical spacing, but also other settings that people might want per-system, like the ability to have a smaller staff size, or to control which staves are hidden.  Quite a lot of new possibilities become available once someone designs a facility for defining how system-level controls should work.

**Comment by oMrSmith** [comment]
> what happens if the layout later changes and 21 is no longer the start of the system?

This is why I had the idea to link such an "indent"-setting to the layout symbol "system break" itself.
Hence, if the system break would get deleted later, the indent would vanishes too, 
instead of becoming a space in the middle of a system without intention.

**Comment by MarcSabatella** [comment]
Yeah, that's an idea I've proposed also.  Simplifies the problem greatly of course, but it does limit the usefulness to cases where you actually have and want explicit layout breaks.  Also, selecting the preceding layout break as a way of setting the initial indent for the following system is considerable less intuitive - and certainly no less work - than simply inserting a horizontal frame in front of the next system directly.

A hack I had considered at one point was to have the action of dragging the initial barline on a system actually a) automatically add a system break at the end of the previous system if not already present, and b) automatically add a horizontal frame and transfer the drag action to that frame.  A similar;r approach could be use for dragging the final barline.  But of course, if we did attach properties to the break, that could take the place of the frame.

Anyhow, definitely room for improvement here, but I'd very much want to see the mechanism chosen be sufficiently general as to cover all the different use cases I mentioned and not seem like half a dozen different hacks.

**Comment by oMrSmith** [comment]
> but I'd very much want to see the mechanism chosen be sufficiently general as to cover all the different use cases I mentioned

Yeah, makes sense...
